src/computer_player.c

```diff
@@ -0,0 +1,604 @@
+// computer algrithm for hard level computer
+//
+// features:
+// - Count cards on the table, record all the played card                                               (x)
+// - Quickly get rid of matched numbers in the hand.                                                    (x)
+// - Keeping at most one Wild Card until the very end.                                                  (x)
+// - use reverse card when you have no card to play                                                     (x)
+// - Play series of similar color cards first, use draw 2 card,                                         (x)
+//   skip card, or reverse card first based on the most color that computer has                         (x)
+// - When playing a Wild Card +4, always change the color based on what the computer has the most       (x)
+// - Try to keep at least one +2 card in your hand.         (x)                                         (x)
+// - Play higher number cards first to get more points      (x)                                         (x)
+// - Note the last card color when players have one card left and avoid to play it.
+//
+// input :
+// - card on the table (for update and play)
+// - if player "UNO"
+// - card drawed from the deck
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "cards_management.h"
+ 
+#define MAX 40
```

Comment:
We have #define MAX_CARDS_NUM  40 in cards_management.h. You can directly use MAX_CARDS_NUM here, and remove the duplicated #define MAX 40